Latest CI build very slow to open some docset

When trying the latest CI build (zeal-unstable-0.2.1-146-gac2225b) I found that when click on some docsets the zeal window just freeze for a long time, probably 20 seconds or longer. At the end it still opens the doc. If switch to another to another doc and back, it has high chance to still freeze for a long time.

I noticed at least docker and font awesome doc have the issue. There is no such problem in 0.2.1. I've redownloaded the two docsets but it is still the same.

I'm trying to compile Zeal against Qt 5.7.0 & new Qt WebKit TP4. But I got below error in QtCreator.

```
linking ..\..\bin\zeal.exe 
Core.lib(extractor.obj) : error LNK2019: unresolved external symbol __imp__archive_read_new referenced in function "public: void __thiscall Zeal::Core::Extractor::extract(class QString const &,class QString const &,class QString const &)" (?extract@Extractor@Core@Zeal@@QAEXABVQString@@00@Z)
Core.lib(extractor.obj) : error LNK2019: unresolved external symbol __imp__archive_read_support_filter_all referenced in function "public: void __thiscall Zeal::Core::Extractor::extract(class QString const &,class QString const &,class QString const &)" (?extract@Extractor@Core@Zeal@@QAEXABVQString@@00@Z)
Core.lib(extractor.obj) : error LNK2019: unresolved external symbol __imp__archive_read_support_format_all referenced in function "public: void __thiscall Zeal::Core::Extractor::extract(class QString const &,class QString const &,class QString const &)" (?extract@Extractor@Core@Zeal@@QAEXABVQString@@00@Z)
Core.lib(extractor.obj) : error LNK2019: unresolved external symbol __imp__archive_read_open_filename referenced in function "public: void __thiscall Zeal::Core::Extractor::extract(class QString const &,class QString const &,class QString const &)" (?extract@Extractor@Core@Zeal@@QAEXABVQString@@00@Z)
Core.lib(extractor.obj) : error LNK2019: unresolved external symbol __imp__archive_read_next_header referenced in function "public: void __thiscall Zeal::Core::Extractor::extract(class QString const &,class QString const &,class QString const &)" (?extract@Extractor@Core@Zeal@@QAEXABVQString@@00@Z)
Core.lib(extractor.obj) : error LNK2019: unresolved external symbol __imp__archive_read_extract referenced in function "public: void __thiscall Zeal::Core::Extractor::extract(class QString const &,class QString const &,class QString const &)" (?extract@Extractor@Core@Zeal@@QAEXABVQString@@00@Z)
Core.lib(extractor.obj) : error LNK2019: unresolved external symbol __imp__archive_read_extract_set_progress_callback referenced in function "public: void __thiscall Zeal::Core::Extractor::extract(class QString const &,class QString const &,class QString const &)" (?extract@Extractor@Core@Zeal@@QAEXABVQString@@00@Z)
Core.lib(extractor.obj) : error LNK2019: unresolved external symbol __imp__archive_read_free referenced in function "public: void __thiscall Zeal::Core::Extractor::extract(class QString const &,class QString const &,class QString const &)" (?extract@Extractor@Core@Zeal@@QAEXABVQString@@00@Z)
Core.lib(extractor.obj) : error LNK2019: unresolved external symbol __imp__archive_filter_bytes referenced in function "private: static void __cdecl Zeal::Core::Extractor::progressCallback(void *)" (?progressCallback@Extractor@Core@Zeal@@CAXPAX@Z)
Core.lib(extractor.obj) : error LNK2019: unresolved external symbol __imp__archive_error_string referenced in function "public: void __thiscall Zeal::Core::Extractor::extract(class QString const &,class QString const &,class QString const &)" (?extract@Extractor@Core@Zeal@@QAEXABVQString@@00@Z)
Core.lib(extractor.obj) : error LNK2019: unresolved external symbol __imp__archive_entry_pathname_w referenced in function "public: void __thiscall Zeal::Core::Extractor::extract(class QString const &,class QString const &,class QString const &)" (?extract@Extractor@Core@Zeal@@QAEXABVQString@@00@Z)
Core.lib(extractor.obj) : error LNK2019: unresolved external symbol __imp__archive_entry_set_pathname referenced in function "public: void __thiscall Zeal::Core::Extractor::extract(class QString const &,class QString const &,class QString const &)" (?extract@Extractor@Core@Zeal@@QAEXABVQString@@00@Z)
Ui.lib(qxtglobalshortcut_win.obj) : error LNK2019: unresolved external symbol __imp__RegisterHotKey@16 referenced in function "private: static bool __cdecl QxtGlobalShortcutPrivate::registerShortcut(unsigned int,unsigned int)" (?registerShortcut@QxtGlobalShortcutPrivate@@CA_NII@Z)
Ui.lib(qxtglobalshortcut_win.obj) : error LNK2019: unresolved external symbol __imp__UnregisterHotKey@8 referenced in function "private: static bool __cdecl QxtGlobalShortcutPrivate::unregisterShortcut(unsigned int,unsigned int)" (?unregisterShortcut@QxtGlobalShortcutPrivate@@CA_NII@Z)
..\..\bin\zeal.exe : fatal error LNK1120: 14 unresolved externals
```

--- a/src/libs/core/application.cpp
+++ b/src/libs/core/application.cpp
@@ -23,6 +23,7 @@
 #include "application.h"
 
 #include "extractor.h"
+#include "networkaccessmanager.h"
 #include "settings.h"
 
 #include <registry/docsetregistry.h>
@@ -35,7 +36,6 @@
 #include <QJsonDocument>
 #include <QJsonObject>
 #include <QMetaObject>
-#include <QNetworkAccessManager>
 #include <QNetworkProxy>
 #include <QNetworkReply>
 #include <QScopedPointer>
@@ -59,7 +59,7 @@ Application::Application(QObject *parent) :
     m_instance = this;
 
     m_settings = new Settings(this);
-    m_networkManager = new QNetworkAccessManager(this);
+    m_networkManager = new NetworkAccessManager(this);
 
     // Extractor setup
     m_extractorThread = new QThread(this);

--- a/src/libs/core/networkaccessmanager.cpp
+++ b/src/libs/core/networkaccessmanager.cpp
@@ -1,0 +1,44 @@
+/****************************************************************************
+**
+** Copyright (C) 2016 Oleg Shparber
+** Contact: https://go.zealdocs.org/l/contact
+**
+** This file is part of Zeal.
+**
+** Zeal is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** Zeal is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Zeal. If not, see <https://www.gnu.org/licenses/>.
+**
+****************************************************************************/
+
+#include "networkaccessmanager.h"
+
+#include <QNetworkRequest>
+
+using namespace Zeal::Core;
+
+NetworkAccessManager::NetworkAccessManager(QObject *parent)
+    : QNetworkAccessManager(parent)
+{
+}
+
+QNetworkReply *NetworkAccessManager::createRequest(QNetworkAccessManager::Operation op,
+                                             const QNetworkRequest &request,
+                                             QIODevice *outgoingData)
+{
+    // Detect URLs without schema, and prevent them from being requested on a local filesytem.
+    const QUrl url = request.url();
+    if (url.scheme() == QLatin1String("file") && !url.host().isEmpty())
+        return QNetworkAccessManager::createRequest(GetOperation, QNetworkRequest(), outgoingData);
+
+    return QNetworkAccessManager::createRequest(op, request, outgoingData);
+}

--- a/src/libs/core/networkaccessmanager.h
+++ b/src/libs/core/networkaccessmanager.h
@@ -1,0 +1,45 @@
+/****************************************************************************
+**
+** Copyright (C) 2016 Oleg Shparber
+** Contact: https://go.zealdocs.org/l/contact
+**
+** This file is part of Zeal.
+**
+** Zeal is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** Zeal is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Zeal. If not, see <https://www.gnu.org/licenses/>.
+**
+****************************************************************************/
+
+#ifndef ZEAL_CORE_NETWORKACCESSMANAGER_H
+#define ZEAL_CORE_NETWORKACCESSMANAGER_H
+
+#include <QNetworkAccessManager>
+
+namespace Zeal {
+namespace Core {
+
+class NetworkAccessManager : public QNetworkAccessManager
+{
+    Q_OBJECT
+public:
+    NetworkAccessManager(QObject *parent = nullptr);
+
+protected:
+    QNetworkReply *createRequest(Operation op, const QNetworkRequest &request,
+                                 QIODevice *outgoingData = nullptr) override;
+};
+
+} // namespace Core
+} // namespace Zeal
+
+#endif // ZEAL_CORE_NETWORKACCESSMANAGER_H

